syntax = "proto3";

import "google/protobuf/any.proto";

package pedro.pb;

enum IpcType {
  REQUEST_RESPONSE = 0;
  FIRE_AND_FORGET = 1;
}

enum MachineArch {
  x86 = 0;
  x64 = 1;
}

message IpcRequest {
  MachineArch arch = 1;
  IpcType type = 2;
  uint64 handle = 3;
  google.protobuf.Any arguments = 4;
}

message IpcResponse {
  bool success = 1;
  google.protobuf.Any value = 2;
}

message Void {

}